### DISCUSSION TITLE
Disable Row Selection in DataTable with Improved "Select All" Button Behavior

### DISCUSSION THREAD

**Discussion by stichingsd-vitrion** [primefaces/community#911] [PrimeVue] [answered]
Hey PrimeVue Team,

I've been using PrimeVue for a while now, and I've got a feature request that I think would be a fantastic addition to DataTables.

I'd love to see the ability to disable the selection of certain rows in a DataTable. I tried to implement this myself, but I ran into a challenge with the "select all" button. It's quite tricky to make it respect the disabled selection on rows.

So, here's my suggestion:

Introduce a feature that allows us to disable row selection in DataTables for specific rows.
Ensure that the "select all" button takes into account the disabled selection on rows, so it behaves intuitively even when some rows are not selectable.
I believe this feature would greatly enhance the usability and flexibility of DataTables in PrimeVue. It would also make it easier for developers like me to achieve the desired behavior without workarounds.

**Comment by mertsincan** [comment, marked as answer]
Hi,

I think this link can help to you;
https://dev.to/mertsincan/handling-big-data-on-datatable-with-checkbox-selection-multiselect-4hgc#checkbox-condition

Note: We can add `isDataSelectable` function in the next versions. I added it to our roadmap.

Best Regards,
